src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceEvaluator.cs

```diff
@@ -0,0 +1,1374 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    static class RecurrenceEvaluator
+    {
+        //
+        // Error Message
+        const string OutOfRange = "The value is out of the accepted range.";
+        const string UnrecognizableValue = "The value is unrecognizable.";
+        const string RequiredParameter = "Value cannot be null.";
+        const string NotMatched = "Start date is not a valid first occurrence.";
+
+        //
+        // Day of week
+        const string Sunday = "Sunday";
+        const string Monday = "Monday";
+        const string Tuesday = "Tuesday";
+        const string Wednesday = "Wednesday";
+        const string Thursday = "Thursday";
+        const string Friday = "Friday";
+        const string Saturday = "Saturday";
+
+        //
+        // Index
+        const string First = "First";
+        const string Second = "Second";
+        const string Third = "Third";
+        const string Fourth = "Fourth";
+        const string Last = "Last";
+
+        //
+        // Recurrence Pattern Type
+        const string Daily = "Daily";
+        const string Weekly = "Weekly";
+        const string AbsoluteMonthly = "AbsoluteMonthly";
+        const string RelativeMonthly = "RelativeMonthly";
+        const string AbsoluteYearly = "AbsoluteYearly";
+        const string RelativeYearly = "RelativeYearly";
+
+        //
+        // Recurrence Range Type
+        const string EndDate = "EndDate";
+        const string Numbered = "Numbered";
+        const string NoEnd = "NoEnd";
+
+        const int WeeklyUnitIntervalDuration = 7; // in days
+        const int MonthlyUnitIntervalDuration = 28; // in days
+        const int YearlyUnitIntervalDuration = 365; // in days
```

Comment:
We probably should make all these const strings public. You have the `RecurrencePattern` public, but not their values.